app/uk/gov/hmrc/testuser/services/Generator.scala

```diff
@@ -38,7 +38,8 @@ class Generator @Inject() extends Randomiser {
     taxOfficeNumber <- Gen.choose(100, 999).map(x => x.toString)
     taxOfficeReference <- Gen.listOfN(10, Gen.alphaNumChar).map(_.mkString.toUpperCase)
   } yield EmpRef.fromIdentifiers(s"$taxOfficeNumber/$taxOfficeReference")
-  private val vrnGenerator = Gen.choose(666000000, 666999999)
+
+  private val vrnGenerator: Gen[String] = Gen.choose(6660000, 6669999).map(i => VrnChecksum.apply(i.toString)).retryUntil(VrnChecksum.isValid)
   private val arnGenerator = new ArnGenerator()
```

Comment:
Can we wrap this in a VrnGenerator class in the same way as ArnGenerator, MtdItIdGenerator, LisaGenerator etc.?

Reply:
What would be the benefit of wrapping this generator?

Reply:
Mainly just to be consistent with the others, but I think it also aids readability, encapsulates the logic and will help with maintainability/extensibility in the future.